bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,333 @@
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    private String getUrl(String type) {
+        String urlStr = type;
+        return urlStr.replace("%IP%", StringUtils.trimToEmpty(config.ip));
+    }
+
+    private void sendData(String key, String value) {
+        String urlStr = getUrl(GoEChargerBindingConstants.MQTT_URL).replace("%KEY%", key).replace("%VALUE%", value);
+        logger.debug("POST URL = {}", urlStr);
+
+        String result = null;
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.POST)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = contentResponse.getContentAsString();
+            // TODO check if value really changed
+            logger.debug("Response: {}", result);
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "No response received on command");
+            return;
+        }
+
+        if (result != null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return;
+        }
+    }
+
+    /**
+     * Request new data from Go-E charger
+     *
+     * @return the Go-E charger object mapping the JSON response or null in case of
+     *         error
+     */
+    @Nullable
+    private GoEStatusResponseDTO getGoEData() {
+        GoEStatusResponseDTO result = null;
+
+        String urlStr = getUrl(GoEChargerBindingConstants.API_URL);
+        logger.debug("GET URL = {}", urlStr);
+
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.GET)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = gson.fromJson(contentResponse.getContentAsString(), GoEStatusResponseDTO.class);
+            // logger.debug("Response: {}", contentResponse.getContentAsString());
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            retryCounter++;
+            if (retryCounter == 1) {
+                logger.warn("Error in getting data from Go-E charger, retrying once");
+                return getGoEData();
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "No response received");
+            result = null;
+        }
+
+        if (result != null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return result;
+        }
+
+        return null;
+    }
+
+    private void refresh() {
+        // Request new GoE data
+        retryCounter = 0;
+        GoEStatusResponseDTO goeResponse = getGoEData();
+
```

Comment:
```suggestion
if (goeResponse == null) {
    allChannels.forEach(channel -> updateState(channel, UndefType.UNDEF));
} else {
   allChannels.forEach(channel -> updateState(channel, getValue(channel, goeResponse)));
}

```

Reply:
Seems to work, will do it like this than. Thanks for your input.